decompress.c

```diff
@@ -65,6 +65,68 @@ static sqfs_err sqfs_decompressor_xz(void *in, size_t insz,
 	return SQFS_OK;
 }
 #define CAN_DECOMPRESS_XZ 1
+
+// The following implementation was adapted from  squashfs-tools
+// https://github.com/plougher/squashfs-tools.git
+
+#define LZMA_PROPS_SIZE 5
+#define LZMA_UNCOMP_SIZE 8
+#define LZMA_HEADER_SIZE (LZMA_PROPS_SIZE + LZMA_UNCOMP_SIZE)
+#define MEMLIMIT (32 * 1024 * 1024)
```

Comment:
This looks very magical

Reply:
for ref: [here](https://asawicki.info/news_1368_lzma_sdk_-_how_to_use.html)
`LZMA_PROPS_SIZE` - `props means a 5-byte header where the library stores some settings. It must be saved with compressed data to be given to the library before decompression.`

`MEMLIMIT` - option to provide max memory size to use.

All other values - don't really dive into it to understand - I just think that it would be reasonable to assume that if it worked for squash-tools - it will be good for this project too.